core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -56,6 +56,15 @@
     private final Optional<Symbol> groupIdSymbol;
     private final List<Symbol> outputs;
 
+    public static AggregationNode simpleSingleAggregation(
```

Comment:
Let's just add simpler `AggregationNode` constructor or rename method to `singleAggregation`

Reply:
`AggregationNodeBuilder` and this method have a different purposes. 'AggregationNodeBuilder' is used to copy existing `AggregationNode` with potentially some properties modified.
`simpleSingleAggregation` creates basic `SINGLE` step `AggregationNode`, without `hashSymbol`, `groupIdSymbol` or `preGroupedSymbols`. That's why I called it _simple_.

Reply:
> AggregationNodeBuilder and this method have a different purposes. 'AggregationNodeBuilder' is used to copy existing AggregationNode with potentially some properties modified.

Builders are used to simplify creation of complex object with multiple arguments. That they can copy values from existing object is an addition

Reply:
This is based on `io.trino.client.ClientSession.Builder` and similar classes. Why do we need a _proper_ builder?

Reply:
Look at `io.trino.cost.PlanNodeStatsEstimate#buildFrom`. There are many more builders like that.

> Why do we need a proper builder?

You've made half-step towards builder, let's just do a full step and be done with it. Then we can avoid `simpleSingleAggregation` and aggregation builder would be usable in other places too.